roles/kubernetes-apps/ansible/templates/kubedns-deploy.yml.j2

```diff
@@ -33,6 +33,9 @@ spec:
 {% endif %}
       nodeSelector:
         beta.kubernetes.io/os: linux
+{% if dns_nodetype is defined %}
+        node-role.kubernetes.io/{{ dns_nodetype }}: ""
```

Comment:
The [node affinity rule ](https://github.com/kubernetes-incubator/kubespray/pull/3746/files#diff-68173b6b58791885bcb383ad8e8f1b85L49) might need to be modified as well.